in_toto/models/metadata.py

```diff
@@ -0,0 +1,207 @@
+"""
+<Program Name>
+  metadata.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+  Santiago Torres <santiago@nyu.edu>
+
+<Started>
+  Oct 23, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides a container class `Metablock` for signed metadata and
+  functions for signing, signature verification, de-serialization and
+  serialization from and to JSON.
+
+"""
+
+import attr
+import json
+import canonicaljson
+
+import securesystemslib.keys
+import securesystemslib.formats
+import securesystemslib.exceptions
+
+from in_toto.models.link import Link
+from in_toto.models.layout import Layout
+from in_toto.exceptions import SignatureVerificationError
+
+@attr.s(repr=False, init=False)
+class Metablock(object):
+  """ This object holds the in-toto metablock data structure. This includes
+  the fields "signed" and "signatures", i.e., what was signed and the
+  signatures. """
+  signatures = attr.ib()
+  signed = attr.ib()
+
+
+  def __init__(self, **kwargs):
+    self.signatures = kwargs.get("signatures", [])
+    self.signed = kwargs.get("signed")
+
+
+  def __repr__(self):
+    """Returns a JSON string representation of the object."""
+    # the double {{'s is the escape sequence for an individual {. We wrap this
+    # under a format string to avoid encoding to json twice (which turns a json
+    # string into a string and so on...
+    # FIXME:
+    # We are mixing 3 JSON string formats here: The value of "signed" is
+    # "pretty printed canonical json", the value of "signatures" is
+    # "canonical json" and the container is just "json".
+    # Is this really what we want?
```

Comment:
@SantiagoTorres, what do you think? I'd say, either we make it all (pretty printed) canonical JSON or none of it.
The only really important thing is that the JSON string of the `signed` object is deterministic when we create or verify signatures.

Reply:
https://github.com/in-toto/in-toto/issues/144